src/command_modules/azure-cli-resource/tests/test_locks.py

```diff
@@ -0,0 +1,39 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import unittest
+
+from azure.cli.testsdk import ScenarioTest, ResourceGroupPreparer
+
+
+class StorageAccountTests(ScenarioTest):
+    def test_list_locks(self):
+        locks_list = self.cmd('az lock list').get_output_in_json()
+        assert len(locks_list) == 0
+
+    def test_lock_create_list_delete(self):
+        for lock_type in ['ReadOnly', 'CanNotDelete']:
+            self.cmd('az lock create -n foo --lock-type {}'.format(lock_type))
+            locks_list = self.cmd('az lock list').get_output_in_json()
+            assert len(locks_list) > 0
+            lock = self.cmd('az lock show -n foo').get_output_in_json()
+            self.assertEqual(lock.get('name', None), 'foo')
+            self.assertEqual(lock.get('level', None), lock_type)
+            self.cmd('az lock delete -n foo')
+
+    @ResourceGroupPreparer()
+    def test_lock_create_list_delete_resource_group(self, resource_group):
+        for lock_type in ['ReadOnly', 'CanNotDelete']:
+            self.cmd('az lock create -n foo -g {} --lock-type {}'.format(resource_group, lock_type))
+            locks_list = self.cmd('az lock list').get_output_in_json()
+            assert len(locks_list) > 0
+            lock = self.cmd('az lock show -g {} -n foo'.format(resource_group)).get_output_in_json()
+            self.assertEqual(lock.get('name', None), 'foo')
+            self.assertEqual(lock.get('level', None), lock_type)
+            self.cmd('az lock delete -g {} -n foo'.format(resource_group))
```

Comment:
I see subscription level locks (I believe) and resource group level locks, but what about resource level locks?  A simple scenario or something like a VNET would be good as well since that's another code path.